internal/webhook/external/azure.go

```diff
@@ -0,0 +1,219 @@
+package external
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/helm"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/image"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	azureSecretDataKey = "secret"
+	azure              = "azure"
+)
+
+func init() {
+	registry.register(
+		azure,
+		webhookReceiverRegistration{
+			predicate: func(cfg kargoapi.WebhookReceiverConfig) bool {
+				return cfg.Azure != nil
+			},
+			factory: newAzureWebhookReceiver,
+		},
+	)
+}
+
+// azureWebhookReceiver is an implementation of WebhookReceiver that handles
+// inbound webhooks from both Azure Container Registry (ACR) and Azure DevOps.
+type azureWebhookReceiver struct {
+	*baseWebhookReceiver
+}
+
+// newAzureWebhookReceiver returns a new instance of azureWebhookReceiver.
+func newAzureWebhookReceiver(
+	c client.Client,
+	project string,
+	cfg kargoapi.WebhookReceiverConfig,
+) WebhookReceiver {
+	return &azureWebhookReceiver{
+		baseWebhookReceiver: &baseWebhookReceiver{
+			client:     c,
+			project:    project,
+			secretName: cfg.Azure.SecretRef.Name,
+		},
+	}
+}
+
+// getReceiverType implements WebhookReceiver.
+func (a *azureWebhookReceiver) getReceiverType() string {
+	return azure
+}
+
+// getSecretValues implements WebhookReceiver.
+func (a *azureWebhookReceiver) getSecretValues(
+	secretData map[string][]byte,
+) ([]string, error) {
+	secretValue, ok := secretData[azureSecretDataKey]
+	if !ok {
+		return nil,
+			errors.New("secret data is not valid for an Azure WebhookReceiver")
```

Comment:
Looks like @hiddeco improved these error messages recently across all the existing receivers. This would make this one match:

```suggestion
			fmt.Errorf("missing data key %q for Azure WebhookReceiver", azureSecretDataKey)
```